### ISSUE TITLE
Block Toolbar: Double focus outline applied to anchor element button

### ISSUE THREAD

**Issue by t-hamano** [WordPress/gutenberg#46757]
### Description

If anchor links are present in the block toolbar, two outlines will be displayed when focused.

For _normal_ button elements, the outer outline (`box-shadow`) is hidden by the `:enabled` pseudo-class:

![button](https://user-images.githubusercontent.com/54422211/209269222-c3b4dd22-5e90-47ea-b94a-03c8cd2db850.png)

However, this selector doesn't apply to anchor elements:

![toolbar](https://user-images.githubusercontent.com/54422211/209269435-767eb371-4b6f-46f5-93dc-d483a50388e5.png)

As I understand it, the `:enabled` selector applies when the following conditions are met:

- The element supports the `disabled` attribute
- The element doesn't have the `disabled` attribute

Since **the anchor element doesn't support the disabled attribute itself**, I assume that this selector doesn't apply.

### Step-by-step reproduction instructions

- Select a `template-part` on the block toolbar.
- Focus on that button with the keyboard controls or click on it with the mouse cursor (Hold the cursor click.).
- Confirm that two outlines are displayed. The outer outline is unexpected.
